### PR TITLE
update/use meta string to scan duplicate

### LINKED ISSUE #109: Move the scanSeconds to the production/development plists
In ScannerViewController move the scanSeconds into a plist constant.

### LINKED ISSUE #110: Change the logic for the time delay between scans to use now() snapshots
The pseudo-code to make this work is in ScannerViewController.ScannerView.swift
```
 //  Last Ticket and Latest Ticket are the same - don't rescan.
            if let scannedTicketID = self.scannerViewModel?.redeemedTicket?.id {
                if ticketID != scannedTicketID || now() > self.lastScannedTicketIdTimer + RESCAN_AFTER_X_SECONDS {
                    self.lastScannedTicketIdTimer = now()
                    self.checkinAutomatically(withTicketID: scannedTicketID, fromGuestTableView: false, atIndexPath: nil)
                    self.stopScanning = false
                    self.timer?.invalidate()
                    self.timer = nil
                    return
                }
            }
```

### LINKED ISSUE #111: Use the metaDataString for duplicate / countdown checks
Cache the metaDataString in the same way we cache the redeemedTicket to compare if the same barcode has been scanned multiple times in a row. Apply the new lastScannedTime method using the metadata instead of the ticketId.

If there is a new metaDataString OR the countdown has expired continue the normal redemption flow with no further checks.

### PR DESCRIPTION
**Pull request by mhlangagc**
### Closes Issues:
- closes #109 
- closes #110 
- closes #111 

### Description:
- Fixes the Infinite scroll problem.

### Pods Installed
 * No change

### CODE REVIEW

File: Big Neon Core/Services/TokenService/TokenService.swift [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+                    return
+                }
+                completion(.tokenRefreshed, nil)
+            }
+            return
+        }
+        
+        do {
+            let jwt = try decode(jwt: accessToken)
+            let _ = jwt.expired
```
[the remark this answers is not in the document]
Comment (mhlangagc) [reply Big Neon Core/Services/TokenService/TokenService.swift]: Old line. It's been removed.

File: Big Neon Core/Services/TokenService/TokenService.swift [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                    return
+                }
+                completion(.tokenRefreshed, nil)
+            }
+            return
+        }
+        
+        do {
+            let jwt = try decode(jwt: accessToken)
+            guard let tokenExpiryDate = jwt.expiresAt else {
```
Comment (Krakaw) [inline Big Neon Core/Services/TokenService/TokenService.swift]:
```suggestion
            guard let tokenExpiryDate = jwt.exp else {
```

File: Big Neon/Screens/Profile/ProfileViewController/ProfileViewController.TableView.swift [hunk 3]
```diff
@@ -31,21 +32,20 @@ extension ProfileViewController {
         case 0:
             return "Account Details"
         default:
-            return "Event Tools"
+            return "Build: 210  ||  Version: 1.0.0"
```
Comment (Krakaw) [inline Big Neon/Screens/Profile/ProfileViewController/ProfileViewController.TableView.swift]: Hardcoded versions?

Comment (mhlangagc) [reply Big Neon/Screens/Profile/ProfileViewController/ProfileViewController.TableView.swift]: Sorted!

**Pull request merged**